documentation/src/docs/asciidoc/user-guide/advanced-topics/engines.adoc

```diff
@@ -78,3 +78,30 @@ For example, the `junit-jupiter-engine` module registers its
 `org.junit.jupiter.engine.JupiterTestEngine` in a file named
 `org.junit.platform.engine.TestEngine` within the `/META-INF/services` folder in the
 `junit-jupiter-engine` JAR.
+
+[[test-engines-minimum-requirements]]
+==== Minimum Requirements
+
+For interoperability with build tools and IDEs, `TestEngine` implementations must adhere
+to the following requirements:
+
+* The `TestDescriptor` returned from `TestEngine.discover()` _must_ be the root of a tree
+  of `TestDescriptor` instances. This implies that there _must not_ be any cycles between
+  a node and its descendants.
+* The root node _should_ have children rather than being completely dynamic. This allows
```

Comment:
I would move _should_ requirements into a section for enhanced compatibility, also listing support for `tags`.

Reply:
Done in 45f8b885f2f35ba5ada6200f5381ba067d800ea1.